src/shell/plugins/maps.c

```diff
@@ -22,6 +22,35 @@
 
 #include "maps.h"
 
+char *maps_lpeers_create (flux_shell_t *shell)
```

Comment:
Seems like another place where a solution to https://github.com/flux-framework/flux-core/issues/3820 would be nice. I like the idea of returning an idset for this purpose. What if we added a `taskids` member to the `flux_shell_get_rank_info(3)` payload:

```c
/*  Return rank and task info for given shell rank as JSON string.
 *  {
 *   "broker_rank":i,
 *   "ntasks":i
 *   "taskids":s  # RFC 22 idset of task ids on this rank
 *   "resources": { "cores":s, ... }
 *  }
 */
int flux_shell_get_rank_info (flux_shell_t *shell,
                              int shell_rank,
                              char **json_str);


```

I'll update flux-framework/flux-core#3820

Reply:
Yup, that would be helpful, also for the `maps_proc_create()` function. 

I would just have to derange the idset, but that is not a problem.